Story/StoryHooks.cs

```diff
@@ -57,14 +58,67 @@ private void StoryHooks()
             On.SLOracleSwarmer.BitByPlayer += SLOracleSwarmer_BitByPlayer;
             On.CoralBrain.CoralNeuronSystem.PlaceSwarmers += OnCoralNeuronSystem_PlaceSwarmers;
             On.SSOracleSwarmer.NewRoom += SSOracleSwarmer_NewRoom;
+            On.HUD.TextPrompt.Update += TextPrompt_Update;
 
+            On.HUD.TextPrompt.UpdateGameOverString += TextPrompt_UpdateGameOverString;
+        }
+
+
+        private void TextPrompt_UpdateGameOverString(On.HUD.TextPrompt.orig_UpdateGameOverString orig, TextPrompt self, Options.ControlSetup.Preset controllerType)
+        {
+            if (OnlineManager.lobby.gameMode is StoryGameMode)
+            {
+                if (!OnlineManager.lobby.isOwner)
+                {
+                    self.gameOverString = "A slugcat has fallen. Perform a rescue, or wait for host to shelter or die";
+
+                }
+
+                if (OnlineManager.lobby.isOwner)
+                {
+                    self.gameOverString = "A slugcat has fallen. Perform a rescue, shelter, or press PAUSE BUTTON to restart";
+
+                }
+            }
+            else
+            {
+                orig(self, controllerType);
+
+            }
+
+        }
+
+        private void TextPrompt_Update(On.HUD.TextPrompt.orig_Update orig, TextPrompt self)
+        {
+            orig(self);
+            if (OnlineManager.lobby.gameMode is StoryGameMode)
+            {
+
+                if (!OnlineManager.lobby.isOwner && self.currentlyShowing == TextPrompt.InfoID.GameOver)
+                {
+                    self.restartNotAllowed = 1; // block clients from GoToDeathScreen
+                    bool touchedInput = false;
+
+                    // let clients still have access to pause menu
+                    for (int j = 0; j < self.hud.rainWorld.options.controls.Length; j++)
+                    {
+                        touchedInput = ((self.hud.rainWorld.options.controls[j].gamePad || !self.defaultMapControls[j]) ? (touchedInput || self.hud.rainWorld.options.controls[j].GetButton(5) || RWInput.CheckPauseButton(0, inMenu: false)) : (touchedInput || self.hud.rainWorld.options.controls[j].GetButton(11)));
+                    }
+                    if (touchedInput)
+                    {
+                        self.gameOverMode = false;
```

Comment:
Hook into gameovermodeActive instead of doing it in an update function

Reply:
or hook into showpausemenu

Reply:
Going to wait for @youbitchoc to address or abandon the following changes they desire to make that I saw in Discord:

```
host dies, someone shelters -> should count as win
host dies, everyone dies too -> should count as loss
```

since that would impact how we currently are discussing changing gameover text menus.

Reply:
Unfortunately, if gameOverMode is not managed here, it creates a few issues. If it is set to false in the ctor, then they will not have the text show up to wait for resuce, shelter etc. They will also lack the power to remove the TextPrompt.

 To that second point, the planned inclusion of [spectating ](https://github.com/henpemaz/Rain-Meadow/pull/210) in the future would benefit from the lack of TextPrompt running as a QoL while watching others. 

PauseMenu_SpawnExitContinueButtons could be used when activated to also remove the gameOverMode so the text prompt is removed, which would only be activated on a user's action versus during Update():

ex:
```
        private void PauseMenu_SpawnExitContinueButtons(On.Menu.PauseMenu.orig_SpawnExitContinueButtons orig, Menu.PauseMenu self)
        {
            self.game.cameras[0].hud.textPrompt.gameOverMode = false;
        }
```

 but then we would be reaching across to another class to alter the variable versus hooking the class directly. 

Let me know what you think.

Reply:
If that's the case I would add in additional modes handled by us. Rather than continuing to leverage gameOverMode. Since technically it's not gameOverMode anymore, it's spectatingMode. This'll give us more granular control on which messages to display

Reply:
Sounds good.

Reply:
I went back and tested this some more:

`if (touchedInput)
                    {
                        self.gameOverMode = false; }` 

if we do not do this, clients cannot access the pause menu when they're dead, which means they cannot quit the game. TextPrompt_Update modes block activation of PauseMenu.

Options:

1. We don't display any TextPrompt for the client on GameOver. Gives them access to PauseMenu to quit.
2. We read-only display TextPrompt, and clients can only exit game after Sleep/Death screen
3. We let clients disable gameOverMode while dead to access PauseMenu. 


I can add a new mode for spectating when that's ready, but for the current problem above, let me know what option you wanna pursue.

Reply:
I was viewing this PR based on the title. The main focus was to prevent players from prematurely ending the cycle and to let the host set that pace.

How we want to handle pausing and text hooks is a much more involved process. Trying to address both of those here will balloon. 

We might not want to actually use the games "pause" menu b/c of game timer behaviors. There's another question on who owns the room after it's previous owner dies. If ownership is maintained, then this will probably further complicate things

Reply:
Fair enough. Will do #2. Just a text prompt update. Host sets pace. We address any future changes in future PR.